src/core/diags/factory_diags.cpp

```diff
@@ -573,6 +575,27 @@ Error Diags::ProcessContinuousWave(uint8_t aArgsLength, char *aArgs[], char *aOu
     return error;
 }
 
+Error Diags::ProcessStream(uint8_t aArgsLength, char *aArgs[], char *aOutput, size_t aOutputMaxLen)
+{
+    Error error = kErrorInvalidArgs;
+
+    VerifyOrExit(otPlatDiagModeGet(), error = kErrorInvalidState);
+    VerifyOrExit(aArgsLength > 0, error = kErrorInvalidArgs);
+
+    if (strcmp(aArgs[0], "start") == 0)
+    {
+        error = otPlatDiagRadioTransmitStream(&GetInstance(), true);
```

Comment:
Can we add this as part of the existing [`diag repeat`](https://github.com/openthread/openthread/tree/main/src/core/diags#diag-repeat-delay-length) command? 
Maybe we can extend it to allow tx of frame with inifnite length (can use zero to indicate this) with no delay?

Reply:
Thanks Abtin for the comment.

I am not sure it is same as continual packet transmission what `diag repeat` does. Purpose of this stream command is to asses the radio's RF performance by putting a radio in mode where it can stream pseudo random (PN9 sequence) bits; it does not pull data from transmit fifos. In this mode, the power amplifier of the radio is generated internally using a pseudo-random (PN9 sequence) bit generator. The primary purpose of this mode is to observe the modulated spectrum without having to provide data for the radio.

Reply:
We also have `diag cw start` which is somehat similar.  Can that be pused?

Reply:
Sorry, I did not get `pused` mean. typo? 

We are using `diag cw` command too. `cw` is for unmodulated carrier-wave/ tone. It is basically a different streaming mode.

If you prefer combining these two diag commands we can do that, but we will have to modify the existing `diag cw` to something like `diag stream <mode>` and <mode> can be `cw` or random PN9 stream or something else. 

Please let me know what you think.

Reply:
Ah sorry! extra `p` I meant "used" :).

I wanted to try to see if we can combine this new function by extending an exiting command (so to make it easier for new folks reading the code) but seems like it is different than others.So ok to add it as a new command.

-----

```
diag stream cw
diag stream random
diag stream stop
```
This sounds nice, but I think we may want to keep old/existing commands (for backward compatibility for folks that may be using them). So we need to keep the `diag cw` commands.

We can add the new function as you proposed as new `diag stream` command.

Reply:
Agree not to break existing CLI. I initially thought to modify it, but I did not for the same reason.

Thanks again for the reviewing the changes.